master/internal/checkpoints/postgres_checkpoints.go

```diff
@@ -54,25 +54,40 @@ func GetModelIDsAssociatedWithCheckpoint(ctx context.Context, ckptUUID uuid.UUID
 	return modelIDs, nil
 }
 
+// ModelInfo is a struct containing info used for locating models.
+type ModelInfo struct {
+	ID      int
+	Version int
+	Name    string
+}
+
 // GetRegisteredCheckpoints gets the checkpoints in
 // the model registrys from the list of checkpoints provided.
-func GetRegisteredCheckpoints(ctx context.Context, checkpoints []uuid.UUID) (map[uuid.UUID]bool, error) {
+func GetRegisteredCheckpoints(ctx context.Context, checkpoints []uuid.UUID) (map[uuid.UUID]ModelInfo, error) {
 	var checkpointIDRows []struct {
-		ID uuid.UUID
+		ID           uuid.UUID
+		ModelID      int
+		ModelVersion int
+		ModelName    string
 	}
 
 	if err := db.Bun().NewRaw(`
-	SELECT DISTINCT(mv.checkpoint_uuid) as ID FROM model_versions AS mv
-	WHERE mv.checkpoint_uuid IN (SELECT UNNEST(?::uuid[]));`,
+	SELECT DISTINCT(mv.checkpoint_uuid) as ID, mv.model_id as model_id, mv.version as model_version, m.name as model_name
+	FROM model_versions AS mv LEFT JOIN models as m on mv.model_id=m.id WHERE mv.checkpoint_uuid
+	IN (SELECT UNNEST(?::uuid[]));`,
```

Comment:
I'm curious why we are doing this query this way and not something more bun-oriented?
```
var checkpointIDRows []struct {
    bun.BaseModel `bun:"table:model_versions"`
    ...
}
...
if err := db.Bun().NewSelect().Model(&checkpointIDRows).
    ColumnExpr("DISTINCT(model_versions.checkpoint_uuid) as ID").
    ColumnExpr("model_versions.model_id as model_id").
    Join("LEFT JOIN models AS m").JoinOn("model_versions.model_id = m.id").
    Where("model_versions.checkpoint_uuid IN (?)", bun.In(checkpoints)).Scan(ctx);
    err != nil {
    ...
}

Reply:
i'm not 100% sure why it was like that originally -- i can change it if you'd like.

Reply:
This is more of a nit than anything else, so no sweat, no change needs to be made at this time, I was just curious if you had context. 

I think eventually doing an audit of our queries across the board to identify pain points could be beneficial but is far outside of the scope of any current work.